dustbin.py

```diff
@@ -0,0 +1,37 @@
+from garbage import Garbage
+from paper_garbage import PaperGarbage
+from plastic_garbage import PlasticGarbage
+from dustbin_content_error import DustbinContentError
+
+
+class Dustbin:
+
+    def __init__(self, color):
+        self.color = color
+        self.paper_content = []
+        self.plastic_content = []
+        self.house_waste_content = []
+
+    def throw_out_garbage(self, garbage):
+        if isinstance(garbage, PlasticGarbage):
+            if garbage.is_clean:
+                self.plastic_content.append(garbage)
+            else:
+                raise DustbinContentError
+
+        elif isinstance(garbage, PaperGarbage):
```

Comment:
Annyit itt megjegyeznék, hogy a Python-ban van *multiple inheritance*, vagyis technikailag egy object lehet pl. egyszerre `PaperGarbage` es `PlasticGarbage` is. Ennek vannak fura mellekhatasai: pl. ha mindketto, de eppen `is_clean=false`, de `is_squeezed=true`, akkor mit csinalsz? Te itt exception-t dobsz, de lehetne ugye a `paper_content`-be is tenni, ha forditott sorrendben nezned. Mindenesetre, szerintem helyes a megoldasod, mivel ez nem volt egyertelmuen specifikalva, tehat barmelyik megoldast el kell fogadniuk.

Ugyanakkor, a *pythonic* modszer ilyen esetekben inkabb az un. *duck typing*. Ez egy metafora, onnan jon, hogy: 

> If it looks like a duck, swims like a duck, and quacks like a duck, then it probably is a duck.

Ez azt jelenti, hogy nem ellenorzod a tipust, es utana probalod vegrehajtani a muveletet, hanem forditva. Pl:
```
if isinstance(x, Duck):
    x.quack()
else:
    raise NotADuckException
```
helyett altalaban a *pythonic* stilus:
```
try:
   x.quack()
except TypeError:
   raise NotADuckException
```
Mindez azert alakult ki igy, mert a Python dinamikus nyelv es alig van benne tipus ellenorzes, minden csak akkor derul ki, pont mielott lefuttatnad az adott sort. Mas, kotottebb nyelvekben (pl. Java) mas megoldasok vannak erre.

Ha duck typing-gal akanank megoldani a feladatot, akkor az kb. igy menne:
```
lazy_user = False
try:
    if garbage.is_squeezed():
        self.paper_content.append(garbage)
        return
    else:
        lazy_user = True
except TypeError:
    pass

try:
    if garbage.is_clean:
        self.plastic_content.append(garbage)
        return
    else:
        lazy_user = True
except TypeError:
    pass

if lazy_user:
    raise DustbinContentError("Don't be lazy, please squeeze or clean the garbage. You could have done it.")

try:
    house_waste_content.append(garbage) if garbage.name
except TypeError:
    raise DustbinContentError("Hey don't throw it away! You don't even know what it is!")
```